hydrolib/core/io/bc/models.py

```diff
@@ -236,6 +291,152 @@ class T3D(ForcingBase):
     _verticalpositiontype_validator = get_enum_validator(
         "verticalpositiontype", enum=VerticalPositionType
     )
+    _timeinterpolation_validator = get_enum_validator(
+        "timeinterpolation", enum=TimeInterpolation
+    )
+
+    @root_validator(pre=True)
+    def _validate_quantityunitpair(cls, values):
+        super()._validate_quantityunitpair(values)
+
+        quantityunitpairkey = "quantityunitpair"
+        quantityunitpairs = values[quantityunitpairkey]
+
+        T3D._validate_that_first_unit_is_time_and_has_no_verticalpositionindex(
```

Comment:
Is it necessary (or style?) to include our own class name here?

Reply:
It is necessary. Since the method that follows is a (private) static method. The alternative would be to add these methods that are currently (private) static as local functions in the root validator function `_validate_quantityunitpair` itself.

I don't have a _strong_ preference for either option, but if I had to pick I'd got with the static methods since it makes the root validator `_validate_quantityunitpair` less clogged and more readable.